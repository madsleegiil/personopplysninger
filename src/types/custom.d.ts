declare module "*.svg" {
  const content: any;
  export default content;
}

declare module "*.png" {
  const content: any;
  export default content;
}

declare module "@navikt/arbeidsforhold" {
  const content: any;
  export default content;
}
