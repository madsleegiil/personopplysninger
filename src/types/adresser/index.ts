import { Boadresse } from "./boadresse";
import { Postadresse } from "./postadresse";
import { Tilleggsadresse } from "./tilleggsadresse";
import { UtenlandskAdresse } from "./utenlandskadresse";
import { Kontaktadresse } from "./kontaktadresse";

export interface GeografiskTilknytning {
  bydel?: string;
  datoFraOgMed?: string;
  kommune?: string;
  enhet?: string;
  land?: string;
}

export interface Adresser {
  boadresse?: Boadresse;
  geografiskTilknytning?: GeografiskTilknytning;
  postadresse?: Postadresse;
  prioritertAdresse?: string;
  kontaktadresse: Kontaktadresse;
  datakilder?: {}[];
}
