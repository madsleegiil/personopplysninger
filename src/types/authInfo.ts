export type AuthInfo =
  | {
      authenticated: false;
    }
  | {
      authenticated: true;
      name: string;
      securityLevel: string;
    };

export interface AuthOidc {
  authenticated: true | false;
}
