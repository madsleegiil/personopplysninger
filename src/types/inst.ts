export interface InstInnslag {
  organisasjonsnummer: string;
  institusjonsnavn: string;
  institusjonstype: string;
  kategori: string;
  startdato: string;
  faktiskSluttdato: string;
  registreringstidspunkt: string;
  maskineltAvsluttet: boolean;
}

export type InstInfo = InstInnslag[];
