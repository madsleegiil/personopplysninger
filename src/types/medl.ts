export interface MedlInnslag {
  unntakId: number;
  ident: string;
  fraOgMed: string;
  tilOgMed: string;
  status: "Avvist" | "Gyldig" | "Uavklart";
  statusaarsak: string;
  dekning: string;
  helsedel: boolean;
  medlem: boolean;
  lovvalgsland: string;
  lovvalg: string;
  grunnlag: string;
}

export type MedlInfo = MedlInnslag[];
