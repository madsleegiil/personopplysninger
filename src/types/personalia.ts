export interface Tlfnr {
  jobb?: string;
  mobil?: string;
  privat?: string;
}

export interface Personalia {
  fornavn?: string;
  etternavn?: string;
  personident?: {
    verdi: string;
    type: string;
  };
  kontonr?: string;
  tlfnr?: Tlfnr;
  spraak?: string;
  epostadr?: string;
  personstatus?: string;
  statsborgerskap?: string;
  foedested?: string;
  sivilstand?: string;
  kjoenn?: string;
  datakilder?: {}[];
}
