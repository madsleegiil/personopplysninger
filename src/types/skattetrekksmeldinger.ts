export interface Skattetrekksmelding {
  organisasjonsnummer: string;
  institusjonsnavn: string;
  institusjonstype: string;
  kategori: string;
  startdato: string;
  faktiskSluttdato: string;
  registreringstidspunkt: string;
}

export type Skattetrekksmeldinger = Skattetrekksmelding[];
