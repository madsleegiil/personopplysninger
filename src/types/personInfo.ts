import { Personalia } from "./personalia";
import { Adresser } from "./adresser";

export interface PersonInfo {
  personalia?: Personalia;
  adresser?: Adresser;
}
