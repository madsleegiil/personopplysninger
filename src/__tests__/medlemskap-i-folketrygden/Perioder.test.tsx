import "@testing-library/jest-dom";
import renderer from "react-test-renderer";
import { IntlProvider } from "react-intl";
import nbMessages from "text/nb";
import Perioder from "../../pages/medlemskap-i-folketrygden/Perioder";
import medlInfo from "../../clients/apiMock/app/fetch/medl-info.json";
import { MedlInfo } from "../../types/medl";

jest.mock("react-router-dom", () => ({
  Link: () => <div />,
  useLocation: () => ({
    pathname: "pathname",
  }),
}));
jest.mock("nav-frontend-js-utils", () => ({
  ...jest.requireActual("nav-frontend-js-utils"),
  guid: () => "Mocked UUID",
}));

jest.mock("nav-frontend-hjelpetekst");

describe("Perioder", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <IntlProvider locale={"nb"} messages={nbMessages}>
          <Perioder medlInfo={medlInfo as unknown as MedlInfo} />
        </IntlProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
