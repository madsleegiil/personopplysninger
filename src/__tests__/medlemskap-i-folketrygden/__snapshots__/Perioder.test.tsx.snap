// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Perioder renders correctly 1`] = `
<div
  className="medl__tabs-innhold"
>
  <div
    className="medl__space"
  >
    <p
      className="typo-normal"
    >
      Hvis du bor og jobber i Norge, er du vanligvis medlem i folketrygden. Perioder med denne type medlemskap vil ikke vises i oversikten. Oversikten er kun aktuell for deg som har
    </p>
  </div>
  <div
    className="medl__space"
  >
    <div
      className="medl__checkbox"
    >
      <img
        alt="Sjekkboksikon"
        className="medl__kalender"
        src="Check.svg"
      />
      <p
        className="typo-normal"
      >
        bodd, jobbet eller studert i utlandet, og du har søkt om medlemskap i folketrygden og fått bekreftet det.
      </p>
    </div>
  </div>
  <div
    className="medl__space"
  >
    <div
      className="medl__checkbox"
    >
      <img
        alt="Sjekkboksikon"
        className="medl__kalender"
        src="Check.svg"
      />
      <p
        className="typo-normal"
      >
        jobbet i Norge, men var med i trygdeordningen i et annet land.
      </p>
    </div>
  </div>
  <div
    className="medl__section"
  >
    <h2
      className="typo-undertittel"
    >
      Ulike situasjoner
    </h2>
  </div>
  <div
    className="medl__space"
  >
    <div
      className="medl__checkbox"
    >
      <img
        alt="Sjekkboksikon"
        className="medl__kalender"
        src="Check.svg"
      />
      <p
        className="typo-normal"
      >
        Flytter du til utlandet for å bo eller jobbe, avhenger medlemskap i folketrygden blant annet av hva du gjør og varigheten på oppholdet.
      </p>
    </div>
  </div>
  <div
    className="medl__space"
  >
    <div
      className="medl__checkbox"
    >
      <img
        alt="Sjekkboksikon"
        className="medl__kalender"
        src="Check.svg"
      />
      <p
        className="typo-normal"
      >
        Norske studenter i utlandet vil vanligvis være pliktige medlemmer i folketrygden, hvis de får støtte fra Lånekassen for å studere på heltid ved et utenlandsk universitet eller en høyskole.
      </p>
    </div>
  </div>
  <div
    className="medl__space"
  >
    <div
      className="medl__checkbox"
    >
      <img
        alt="Sjekkboksikon"
        className="medl__kalender"
        src="Check.svg"
      />
      <p
        className="typo-normal"
      >
        Er du bosatt i et annet land, men jobber helt eller delvis i Norge, kan du være med i trygdeordningen i bostedslandet ditt og være unntatt fra medlemskap i folketrygden.
      </p>
    </div>
  </div>
  <div
    className="medl__section"
  >
    <h2
      className="typo-undertittel"
    >
      Hva betyr medlemskap i folketrygden?
    </h2>
  </div>
  <div
    className="medl__space"
  >
    <p
      className="typo-normal"
    >
      Medlemskap i folketrygden betyr at du har rettigheter og plikter i folketrygden. 
      <a
        className="lenke"
        href="/no/person/flere-tema/arbeid-og-opphold-i-norge/relatert-informasjon/medlemskap-i-folketrygden"
      >
         Les om medlemskap i folketrygden
      </a>
      . Kontakt NAV hvis du har spørsmål om perioder med eller uten medlemskap i folketrygden, telefon (+47) 21 07 37 00.
    </p>
  </div>
  <div
    className="medl__space"
  >
    <p
      className="typo-normal"
    >
      Kontakt Statens lånekasse for utdanning hvis du har spørsmål om perioder med støtte fra Lånekassen.
    </p>
  </div>
  <div
    className="medl__section"
  >
    <h2
      className="typo-systemtittel"
    >
      Oversikt over perioder
    </h2>
  </div>
  <div
    className="medl__space"
  >
    <div
      className="alertstripe alertstripe--info"
    >
      <span
        className="alertstripe__ikon"
      >
        <span
          className="sr-only"
        >
          info
        </span>
        <svg
          aria-label="info-ikon"
          focusable="false"
          height="1.5em"
          kind="info-sirkel-fyll"
          role="img"
          viewBox="0 0 24 24"
          width="1.5em"
        >
          <g
            fill="none"
          >
            <path
              d="M12 0C5.382 0 0 5.382 0 12s5.382 12 12 12c6.617 0 12-5.382 12-12S18.617 0 12 0z"
              fill="#337C9B"
            />
            <path
              d="M12 5a1.566 1.566 0 1 1 .11 3.13A1.566 1.566 0 0 1 12 5zm2.976 12.01c.563 0 1.043.431 1.043.991s-.48.992-1.043.992H9.39c-.564 0-1.043-.431-1.043-.992 0-.56.479-.99 1.043-.99h1.6v-5.016h-.986c-.565 0-1.044-.43-1.044-.991 0-.56.48-.991 1.044-.991h2.03c.563 0 1.043.43 1.043.99v6.007h1.899z"
              fill="#FFF"
            />
          </g>
        </svg>
      </span>
      <div
        className="typo-normal alertstripe__tekst"
      >
        Oversikten viser perioder fra og med 2015. Hvis perioden i 2015 henger sammen med en periode fra tidligere år, vil også den vises.
      </div>
    </div>
  </div>
  <div
    className="ekspanderbartPanel medl__space ekspanderbartPanel--lukket ekspanderbartPanel--border"
  >
    <button
      aria-controls="0963719029-00912-05058-9365-6609205610604"
      className="ekspanderbartPanel__hode"
      id="1418810825-2052-06958-7137-22437230626995"
      onClick={[Function]}
      onKeyDown={[Function]}
      type="button"
    >
      <div
        className="ekspanderbartPanel__flex-wrapper"
      >
        <span
          className="ekspanderbartPanel__tittel"
        >
          Med medlemskap
        </span>
        <span
          className="ekspanderbartPanel__indikator"
        />
      </div>
    </button>
    <div
      aria-labelledby="1418810825-2052-06958-7137-22437230626995"
      id="0963719029-00912-05058-9365-6609205610604"
      role="region"
    />
  </div>
  <div
    className="ekspanderbartPanel medl__space ekspanderbartPanel--lukket ekspanderbartPanel--border"
  >
    <button
      aria-controls="994528511-0785-29705-26833-1673297017292"
      className="ekspanderbartPanel__hode"
      id="060296419-2894-6971-18071-6581268817265"
      onClick={[Function]}
      onKeyDown={[Function]}
      type="button"
    >
      <div
        className="ekspanderbartPanel__flex-wrapper"
      >
        <span
          className="ekspanderbartPanel__tittel"
        >
          Uten medlemskap
        </span>
        <span
          className="ekspanderbartPanel__indikator"
        />
      </div>
    </button>
    <div
      aria-labelledby="060296419-2894-6971-18071-6581268817265"
      id="994528511-0785-29705-26833-1673297017292"
      role="region"
    />
  </div>
</div>
`;
